### PR TITLE
Update documentation for `brew livecheck`

### PR DESCRIPTION
**Pull request by mohammadzainabbas**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

#

This PR aims to update the documentation for `brew livecheck` (after https://github.com/Homebrew/brew/pull/13613 is merged).

@samford and @nandahkrishna, kindly have a look and let me know if we need to update some other documentation as well.

### CODE REVIEW

**Review by samford — COMMENTED** [review]
I ran out of time today but I will go through this tomorrow (Wednesday) when I have a chance.

However, just from a cursory glance, I see that you've converted the numbered lists from generic 1s to 1, 2, 3, etc. We intentionally use 1 for each entry because Markdown will automatically turn them into 1, 2, 3, etc. as part of rendering. With that setup, we don't have to worry about manually keeping the numbers up to date as we make changes, which also simplifies diffs. Please return these back to the previous state (i.e., uniformly using `1.`).

**Comment by mohammadzainabbas** [comment]
> We intentionally use 1 for each entry because Markdown will automatically turn them into 1, 2, 3, etc. as part of rendering. With that setup, we don't have to worry about manually keeping the numbers up to date as we make changes, which also simplifies diffs. 

Oh, I didn't know this. Thanks for pointing out.

> Please return these back to the previous state (i.e., uniformly using `1.`).

Done ✅

**Review by samford — CHANGES_REQUESTED** [review]
From my perspective, the changes required in this documentation with respect to resource support are very minimal. The following feedback is focused on paring down the changes in this PR to only what's necessary.

File: docs/Brew-Livecheck.md [hunk 1]
```diff
@@ -1,23 +1,25 @@
 # `brew livecheck`
 
-The `brew livecheck` command finds the newest version of a formula or cask's software by checking upstream. Livecheck has [strategies](https://rubydoc.brew.sh/Homebrew/Livecheck/Strategy.html) to identify versions from various sources, such as Git repositories, websites, etc.
+The `brew livecheck` command finds the newest version of a formula, cask or resource's software by checking upstream. Livecheck has [strategies](https://rubydoc.brew.sh/Homebrew/Livecheck/Strategy.html) to identify versions from various sources, such as Git repositories, websites, etc.
```
[the remark this answers is not in the document]
Comment (mohammadzainabbas) [reply docs/Brew-Livecheck.md:3]: Oh got it.

File: docs/Brew-Livecheck.md [hunk 2]
```diff
@@ -1,23 +1,25 @@
...
 
-This can be accomplished by adding a `livecheck` block to the formula/cask. For more information on the available methods, please refer to the [`Livecheck` class documentation](https://rubydoc.brew.sh/Livecheck.html).
+This can be accomplished by adding a `livecheck` block to the formula/cask/resource. For more information on the available methods, please refer to the [`Livecheck` class documentation](https://rubydoc.brew.sh/Livecheck.html).
 
 ## Creating a check
 
-1. **Use the debug output to understand the situation**. `brew livecheck --debug <formula>|<cask>` provides information about which URLs livecheck tries, any strategies that apply, matched versions, etc.
+1. **Use the debug output to understand the situation**. `brew livecheck --debug <formula>|<cask>` provides information about which URLs livecheck tries, any strategies that apply, matched versions, etc. `brew livecheck --debug --resources <formula>` will provides livecheck's debug information about the resources as well.
+
+> Note: `--resources` flag only works with Formulae.
```
[the remark this answers is not in the document]
Comment (mohammadzainabbas) [reply docs/Brew-Livecheck.md:22]: Okay, reverted ✅

File: docs/Brew-Livecheck.md [hunk 3]
```diff
@@ -1,19 +1,19 @@
 # `brew livecheck`
 
-The `brew livecheck` command finds the newest version of a formula or cask's software by checking upstream. Livecheck has [strategies](https://rubydoc.brew.sh/Homebrew/Livecheck/Strategy.html) to identify versions from various sources, such as Git repositories, websites, etc.
+The `brew livecheck` command finds the newest version of a formula or cask's software software by checking upstream. Livecheck has [strategies](https://rubydoc.brew.sh/Homebrew/Livecheck/Strategy.html) to identify versions from various sources, such as Git repositories, websites, etc.
```
Comment (EricFromCanada) [inline docs/Brew-Livecheck.md:3]:
```suggestion
The `brew livecheck` command finds the newest version of a formula or cask's software by checking upstream. Livecheck has [strategies](https://rubydoc.brew.sh/Homebrew/Livecheck/Strategy.html) to identify versions from various sources, such as Git repositories, websites, etc.
```

**Review by EricFromCanada — CHANGES_REQUESTED** [review]

File: docs/Brew-Livecheck.md (hunk 3 above)
Comment (mohammadzainabbas) [reply docs/Brew-Livecheck.md]: Oops! Thanks for pointing it out.

**Review by samford — APPROVED** [review]
Thanks, @mohammadzainabbas!

**Pull request merged**